tests/framework/db/mssql/QueryBuilderTest.php

```
@@ -419,376 +533,708 @@ public static function upsertProvider(): array
+            IF @constraintName IS NULL BREAK
+            EXEC (N'ALTER TABLE ' + @tableName + N' DROP CONSTRAINT [' + @constraintName + N']')
+        END
+        ALTER TABLE {{foo1}} ADD CONSTRAINT [[UQ_foo1_bar]] UNIQUE ([[bar]])
+        SQL;
+
+        self::assertSame(
+            $expected,
+            $qb->alterColumn('foo1', 'bar', $this->string(30)->unique()),
+            "ALTER COLUMN to 'nvarchar(30)' should drop default constraints and add a UNIQUE constraint.",
+        );
     }
```

PHPMD (2.15.0)

[warning] The method testAlterColumn() has 358 lines of code. Current threshold is set to 100. Avoid really long methods. (undefined)

(ExcessiveMethodLength)